WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderImageScanner.java

```diff
@@ -105,8 +105,21 @@ public String getLargestImage(int minImageWidth) {
             if (width > currentMaxWidth) {
                 currentImageUrl = imageUrl;
                 currentMaxWidth = width;
-            } else if (currentImageUrl == null && hasSuitableClassForFeaturedImage(imageTag)) {
-                currentImageUrl = imageUrl;
+            } else {
+                // Inspect the image tag and look for clues for an image of the right size
+
+                // Look through the srcset attribute if set for the image best matching minImageWidth
+                currentImageUrl = ReaderHtmlUtils.getSrcsetImageForTag(imageTag, minImageWidth);
+
+                // Check if the image tag's class suggests it's a good enough size
+                if (currentImageUrl == null && hasSuitableClassForFeaturedImage(imageTag)) {
+                    currentImageUrl = imageUrl;
+                }
+
+                // Look for a data-large-file attribute if set and use the associated url
+                if (currentImageUrl == null) {
+                    currentImageUrl = ReaderHtmlUtils.getLargeFileAttr(imageTag);
```

Comment:
I think we shouldn't rely on the `data-large-file` attribute, because it's arbitrary - we have no way of knowing what the actual width the image is short of downloading it and seeing. So there's no guarantee that it won't actually be smaller than the `width` attribute if it's set. So I think it makes sense to keep it as a fallback, if we aren't able to pin down a URL with an actual known width.

I'll make a top-level comment laying out how I think the logic of this function should play out overall.

Reply:
I see, thanks for the clarification ;)